### ISSUE TITLE
Get pytest working with SLURM

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo-loihi#9]
In gitlab by @tbekolay on Jun 20, 2018, 09:58

This came up in !15 but isn't critical to fix in that PR. Making this issue so I don't forget to do it once that PR is merged.

**Comment by tbekolay** [comment]
In gitlab by @tbekolay on Jun 21, 2018, 21:30

I'm not sure if this wasn't working before because of something to do with the compiled files? But now on the Intel host I can do

```bash
SLURM=1 pytest --plots
```

and the tests run.

I added a user-wide `matplotlibrc` (meaning it will affect all of us) so that you don't get issues with being in a headless environment. With that change, the plots are generated through `pytest` no problem and look the same as running on the simulator locally.

The only issue is that the test suite will appear to hang after a while. Looking at the output with `pytest -s`, there's a pretty consistent pattern. Two models will run immediately, without hitting the SLURM queue, then the third model is queued:

```bash
...
Building CxGroup(<Connection from <Ensemble "a"> to <Probe of 'decoded_output' of <Ensemble "a">>>) on core.id=7
SLURM is being run in background
srun: job 1923 queued and waiting for resources
Waiting For Connection...
Waiting For Connection...
Waiting For Connection...
...
```

and it never seems to get out of the queue. Killing the process with Ctrl+C and rerunning that specific test will often result in the job running immediately (but not always), so it feels like something about the environment is not set up for running lots of models in quick succession.

Not sure what the best resolution is to that. Could try a few quick hacks, like adding short delays between each test, or timing out and restarting the test after we're in the queue for a specified amount of time.

**Comment by tbekolay** [comment]
In gitlab by @tbekolay on Aug 7, 2018, 11:11

The original issue raised here is fixed, though we still cannot get through the whole test suite due to some state sticking around in NxSDK (running out of spike counters, if I recall correctly). I believe that is tracked in a separate issue though, so I'm going to close this one.

**Comment by tbekolay** [comment]
In gitlab by @tbekolay on Aug 7, 2018, 11:11

closed
